modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/ScriptElementAttributes.tsx

```diff
@@ -0,0 +1,279 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useState} from 'react';
+import {v4 as uuidv4} from 'uuid';
+
+const booleanValue = [
+	{label: 'True', value: true},
+	{label: 'False', value: false},
+];
+
+const dataTypes = ['String', 'Boolean'];
+
+const AttributeFields = ({
+	attribute = '',
+	handleAddClick,
+	handleRemoveClick,
+	id,
+	index,
+	portletNamespace,
+	updateAttributeList,
+	value = '',
+}) => {
+	const [attrErrorMessage, setAttrErrorMessage] = useState<string>('');
+	const [attributeName, setAttributeName] = useState<string>(attribute);
+	const [attributeType, setAttributeType] = useState<string>(dataTypes[0]);
+	const [attributeValue, setAttributeValue] = useState<boolean | string>(
+		value
+	);
+
+	const handleAttributeChange = ({target}) => {
+		const newValue = target.value;
+		setAttributeName(newValue);
+
+		if (newValue.toLowerCase().includes('src')) {
+			setAttrErrorMessage(
+				Liferay.Language.get('use-the-javaScript-url-field')
+			);
+		}
+		else {
+			setAttrErrorMessage('');
+		}
+	};
+
+	useEffect(() => {
+		updateAttributeList(index, {
+			attribute: attributeName,
+			id,
+			type: attributeType,
+			value: attributeValue,
+		});
+	}, [
+		attributeName,
+		attributeType,
+		attributeValue,
+		id,
+		index,
+		updateAttributeList,
+	]);
+
+	useEffect(() => {
+		if (attributeType === 'String') {
+			setAttributeValue('');
+		}
+		if (attributeType === 'Boolean') {
+			setAttributeValue(booleanValue[0].value);
+		}
+	}, [attributeType]);
```

Comment:
Well, normally this is a problem because you have two states that need to synchronize and this can cause problems, mainly making the implementation a little more complex, causing double rendering when the component's local state is updated and later when the parent's state is updated.

This resorts to having just one source of truth or [SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth), it has the same idea here in React.js this makes things simpler. For example, when you need the local state to be used in the parent component, the ideal is to remove the local state and move the state to the parent. In this case, in this component you would consume the data directly from the props instead of adding it to a local state and use the `updateAttributeList` directly in the input's `onChange` calls instead of adding it to `useEffect`.

You could also use Formik here to understand more about how this works but as this is quite simple I would just go with these recommendations that I provided are sufficient because you already have the state in the parent.

Another option here would be to keep the local state and remove the parent's state and create a `<input name="name" hidden />` for each element with the same name, this will arrive in the backend as an array, if I'm not mistaken, you have to check it but it would have almost the same behavior as you are doing in taking the state and adding a string with `filteredList`.

> This second option won't work because I realized that you have control over the lists, so this second option doesn't make sense here in this use case, but it would be a potential option if you don't have control over removing or adding.

Reply:
Thanks! I'll go through your recommendations to find a better solution for this.

Reply:
Hey, @matuzalemsteles! 
Evan gave me a big help and I was able to refactor the code and remove the useEffects. I'll be sending the updated version once I fix the other requests.